wkhtmltopdf-image/src/main/docker/app.py

```diff
@@ -0,0 +1,118 @@
+#! /usr/bin/env python
+import StringIO
+import json
+from subprocess import Popen, PIPE
+
+from werkzeug.exceptions import HTTPException, NotFound
+from werkzeug.routing import Map, Rule
+from werkzeug.wrappers import Request, Response
+
+"""
+This python server is responsible for generating PDFs from HTML
+using an underlying wkhtmltopdf command-line tool.
+
+To use this application, the user must send a POST request with
+base64 or form encoded encoded HTML content and the wkhtmltopdf Options in
+request data, with keys 'contents' and 'options'.
+The application will return a response with the PDF file.
+"""
+class WkHtmlToPdfServer():
+    def __init__(self):
+        self.url_map = Map([
+            Rule('/', endpoint='generate_pdf'),
+            Rule('/health', endpoint='health_check')
+        ])
+
+    """
+    Generate a PDF from submitted HTML.
+    """
+    def on_generate_pdf(self, request):
+        if request.method != 'POST':
+            return
+
+        request_is_json = request.content_type.endswith('json')
+        sourceStream = None
+        options = None
+
+        if request_is_json:
+            # If a JSON payload is there, all data is in the payload
+            payload = json.loads(request.get_data(parse_form_data=True, cache=False))
+            sourceStream = StringIO.StringIO(payload['contents'].decode('base64'))
+            options = payload.get('options', {})
+        elif request.files:
+            # First check if any files were uploaded
+            sourceStream = request.files['file'].stream
```

Comment:
I'm hoping that this will actually stream under the covers but may or may not depending upon how the lower-level libraries are implemented.